subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c

```diff
@@ -413,22 +421,26 @@ fs_mgmt_file_hash_checksum(struct mgmt_ctxt *ctxt)
 
 		if (group->byte_string == true) {
 			/* Output is a byte string */
-			ok &= zcbor_bstr_encode_ptr(zse, output, group->output_size);
+			ok &= zcbor_bstr_encode_ptr(zse, output.str, group->output_size);
 		} else {
 			/* Output is a number */
 			uint64_t tmp_val = 0;
 
 			if (group->output_size == sizeof(uint8_t)) {
-				tmp_val = (uint64_t)(*(uint8_t *)output);
+				tmp_val = (uint64_t)output.u8;
 			} else if (group->output_size == sizeof(uint16_t)) {
-				tmp_val = (uint64_t)(*(uint16_t *)output);
+				tmp_val = (uint64_t)output.u16;
 			} else if (group->output_size == sizeof(uint32_t)) {
-				tmp_val = (uint64_t)(*(uint32_t *)output);
+				tmp_val = (uint64_t)output.u32;
+#if FS_MGMT_CHECKSUM_HASH_LARGEST_OUTPUT_SIZE >= 8
 			} else if (group->output_size == sizeof(uint64_t)) {
-				tmp_val = (*(uint64_t *)output);
+				tmp_val = (uint64_t)output.u64;
+#endif
 			} else {
-				LOG_ERR("Unable to handle numerical checksum size %u",
-					group->output_size);
+				LOG_ERR("Unable to handle numerical checksum size %u, "
+					"max allowed %u", group->output_size,
+					MAX(sizeof(uint64_t),
+					    FS_MGMT_CHECKSUM_HASH_LARGEST_OUTPUT_SIZE));
```

Comment:
This is not correct, for sha256 this is 32, but that is 32 data bytes not a 32 byte integer, and the max of 8 and 32 would return 32, which is not correct as the maximum numerical size would be 4, which is for crc32

Reply:
Yes. This is supposed to be the same as in line 336, MIN.

Reply:
Doesn't look fixed?